src/Components/TodoList.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+
+type Props = {
+  todos: Todo[];
+  onDelete: (id: number) => void;
+  setTodos: (value: Todo[]) => void;
+};
+
+export const TodoList: React.FC<Props> = ({ onDelete, todos, setTodos }) => {
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleStartEditing = (todoId: number, title: string) => {
+    setEditingTodoId(todoId);
+    setNewTitle(title);
+  };
+
+  const handleSaveEditing = (todoId: number, title: string) => {
+    if (!title.trim()) {
+      onDelete(todoId);
+    } else {
+      const updatedTodos = todos.map((todo) => {
+        if (todo.id === todoId) {
+          return { ...todo, title };
+        }
+
+        return todo;
+      });
+
+      setTodos(updatedTodos);
+      setEditingTodoId(null);
+      setNewTitle('');
+    }
+  };
+
+  return (
+    <section className="todoapp__main">
+      {todos.map(({ id, completed, title }) => {
+        const isEditing = editingTodoId === id;
+
+        return (
+          <div
+            key={id}
+            className={classNames('todo', { completed, editing: isEditing })}
+          >
+            <label className="todo__status-label">
+              <input
+                type="checkbox"
+                className="todo__status"
+                checked={completed}
+                onChange={() => {
+                  const updatedTodos = todos.map((todo) => {
+                    if (todo.id === id) {
+                      return { ...todo, completed: !completed };
+                    }
+
+                    return todo;
+                  });
+
+                  setTodos(updatedTodos);
+                }}
```

Comment:
logic should be moved into separate handler instead on anonymous function in jsx.
Fix all similar cases